packages/react/src/SensitiveTextarea/SensitiveTextarea.tsx

```diff
@@ -0,0 +1,53 @@
+import { ActionIcon, Flex, Textarea, TextareaProps } from '@mantine/core';
+import { useClipboard } from '@mantine/hooks';
+import { showNotification } from '@mantine/notifications';
+import { IconCopy } from '@tabler/icons-react';
+import { RefAttributes, useRef, useState } from 'react';
+
+export interface SensitiveTextareaProps extends TextareaProps, RefAttributes<HTMLTextAreaElement> {}
+
+export function SensitiveTextarea(props: Omit<SensitiveTextareaProps, 'ref'>): JSX.Element {
+  const [revealed, setRevealed] = useState(false);
+  const clipboard = useClipboard();
+  const ref = useRef<HTMLTextAreaElement | null>(null);
+
+  const styles: SensitiveTextareaProps['styles'] = {
+    ...props.styles,
+  };
+  if (!revealed) {
+    if (!styles.input) {
+      styles.input = {};
+    }
+
+    styles.input['-webkit-text-security'] = 'disc';
+  }
+
+  return (
+    <Flex gap="xs">
+      <Textarea
+        {...props}
+        styles={{
+          ...styles,
+          root: {
+            ...(styles.root ?? {}),
+            flexGrow: 1,
+          },
+        }}
+        ref={ref}
+        autosize
+        minRows={1}
+        onFocus={() => setRevealed(true)}
+        onBlur={() => setRevealed(false)}
```

Comment:
I think it would be better security practice to have an explicit show/hide control here, allowing people to securely type in a value without someone should surfing

Reply:
That is how it worked in the original PR, but it creates a pretty confusing user experience.

This implementation is how AWS Secrets and AWS Parameter work, which feels more natural.

I'm open to discussion though.

Reply:
Mmkay, I think this way is definitely much better than nothing — good to merge as is

Reply:
Weighing in here: 
Since this is an _editing_ compoent,  then I agree with the current implementation. Having a separate button when I'm _already_ in editing mode. Many UX implementations in the wild don't even hide secrets when editing.

I think the explicit show/hide makes more sense for a _display_ component, which is the more common practice